packages/contracts/contracts/StateChannelTransaction.sol

```diff
@@ -48,16 +45,15 @@ contract StateChannelTransaction {
       "App is not finalized yet"
     );
 
-    Transfer.Transaction memory txn = appRegistry.getResolution(
+    bytes memory resolution = appRegistry.getResolution(
       appIdentityHash
     );
 
-    require(
-      Transfer.meetsTerms(txn, terms),
-      "Transfer details do not meet terms"
-    );
-
-    txn.execute();
+    bytes memory payload = abi.encodeWithSignature(
+      "interpret(bytes,bytes)", resolution, interpreterParams);
+    (bool success, bytes memory returnData) =
+      interpreterAddress.delegatecall(payload);
```

Comment:
cc @chriseth, kind of an interesting use case.

Reply:
Not sure what you mean - what should it represent?

Reply:
Is there any syntax for something like `Interpreter(...).interpret(...).delegatecall()`?

Reply:
`Interpreter(addr).interpret(p1, p2).delegatecall()`would delegatecall the `interpret` function of `addr` using the `Interpreter` contract interface, similarly to how `Interpreter(addr).interpret(p1, p2)` calls the `interpret` function of `addr` using the `Interpreter` contract interface